test/sds/envoy-sds-template.yaml

```diff
@@ -0,0 +1,484 @@
+            "tag_name": "http_conn_manager_listener_prefix",
+            "regex": "^listener(?=\\.).*?\\.http\\.(((?:[_.[:digit:]]*|[_\\[\\]aAbBcCdDeEfF[:digit:]]*))\\.)"
+          },
+          {
+            "tag_name": "http_conn_manager_prefix",
+            "regex": "^http\\.(((?:[_.[:digit:]]*|[_\\[\\]aAbBcCdDeEfF[:digit:]]*))\\.)"
+          },
+          {
+            "tag_name": "listener_address",
+            "regex": "^listener\\.(((?:[_.[:digit:]]*|[_\\[\\]aAbBcCdDeEfF[:digit:]]*))\\.)"
+          },
+          {
+            "tag_name": "mongo_prefix",
+            "regex": "^mongo\\.(.+?)\\.(collection|cmd|cx_|op_|delays_|decoding_)(.*?)$"
+          }
+        ],
+        "stats_matcher": {
+          "inclusion_list": {
+            "patterns": [
+              {{- range $a, $s := .inclusionPrefix }}
+              {
+              "prefix": "{{$s}}"
+              },
+              {{- end }}
+              {{- range $a, $s := .inclusionSuffix }}
+              {
+              "suffix": "{{$s}}"
+              },
+              {{- end }}
+              {{- range $a, $s := .inclusionRegexps }}
+              {
+              "regex": "{{js $s}}"
+              },
+              {{- end }}
+            ]
+          }
+        }
+      },
+      "admin": {
+        "access_log_path": "/dev/null",
+        "address": {
+          "socket_address": {
+            "address": "{{ .localhost }}",
+            "port_value": {{ .config.ProxyAdminPort }}
+          }
+        }
+      },
+      "dynamic_resources": {
+        "lds_config": {
+          "ads": {}
+        },
+        "cds_config": {
+          "ads": {}
+        },
+        "ads_config": {
+          "api_type": "GRPC",
+          "grpc_services": [
+            {
+              "envoy_grpc": {
+                "cluster_name": "xds-grpc"
+              }
+            }
+          ]
+        }
+      },
+      "static_resources": {
+        "clusters": [
+          {
+            "name": "prometheus_stats",
+            "type": "STATIC",
+            "connect_timeout": "0.250s",
+            "lb_policy": "ROUND_ROBIN",
+            "hosts": [
+              {
+                "socket_address": {
+                  "protocol": "TCP",
+                  "address": "{{ .localhost }}",
+                  "port_value": {{ .config.ProxyAdminPort }}
+                }
+              }
+            ]
+          },
+          {
+            "name": "xds-grpc",
+            "type": "STRICT_DNS",
+            "dns_refresh_rate": "{{ .dns_refresh_rate }}",
+            "dns_lookup_family": "{{ .dns_lookup_family }}",
+            "connect_timeout": "{{ .connect_timeout }}",
+            "lb_policy": "ROUND_ROBIN",
+            {{ if eq .config.ControlPlaneAuthPolicy 1 }}
+            "tls_context": {
+              "common_tls_context": {
+                "alpn_protocols": [
+                  "h2"
+                ],
+                {{ if 0 }}
+                {{ if 0 }}
```

Comment:
i assume this is WIP?

Reply:
Partially WIP - it's intended to test file-based SDS cert rotation, but that's manual.
I can remove it from the PR if you want.